Expose catchup_period_seconds in /info JSON (Closes #1176)

alienx5499 opened the pull request:
### Summary
- Surfaces the node’s configured catchup sleep as `catchup_period_seconds` in the `/info` JSON. This helps clients, relays, dashboards, and SREs reason about expected pacing during catchup without guessing from network behavior.

### Motivation
- Networks can configure `Period` and `CatchupPeriod` differently across deployments (e.g., mainnet vs. various testnets). The latter affects how quickly nodes progress when catching up. Since `CatchupPeriod` is not part of the group hash and can vary between nodes, making it visible via `/info` increases transparency and improves tooling.

### Changes
- `common/chain/info.go`
  - Add `CatchupPeriod time.Duration` field to `Info` and populate it in `NewChainInfo` from `key.Group.CatchupPeriod`.
- `common/chain/convert.go`
  - Update `Info.ToJSON` to inject a new JSON field `catchup_period_seconds` (as an unsigned integer) derived from `Info.CatchupPeriod.Seconds()`.
  - Keep proto conversions backward compatible; JSON emission is augmented without requiring immediate proto regeneration.
- `protobuf/drand/common.proto`
  - Add `uint32 catchup_period_seconds = 8;` to `ChainInfoPacket` for forward compatibility when protobufs are regenerated in future work.
- `common/chain/info_test.go`
  - Extend tests to validate that `/info` JSON includes the `catchup_period_seconds` field and that it reflects the group’s `CatchupPeriod`.

### API (JSON) Example
```json
{
  "public_key": "<hex>",
  "period": 3,
  "genesis_time": 1692803367,
  "genesis_seed": "<hex>",
  "chain_hash": "<hex>",
  "scheme": "bls-unchained-g1-rfc9380",
  "beacon_id": "default",
  "catchup_period_seconds": 1
}
```

### Backward Compatibility
- This is an additive JSON field only. No runtime behavior changes. Existing clients remain unaffected and can safely ignore unknown fields. Consumers that care about catchup behavior can start reading the new field.

### Performance & Security
- Performance impact is negligible (a small JSON map injection after marshaling the proto struct). No sensitive data is exposed by this field.

### How To Verify
- Run targeted tests: `go test ./common/chain ./handler/http`.
- Start a node locally and request `/info`; confirm the response includes `catchup_period_seconds` with the expected value.

### Risks / Rollout
- Low risk. The change is purely informational. Relays and proxies that pass through JSON will include the new field transparently.

### Related Issue
- Closes #1176: "Add catchup sleep time to the /info endpoint".

### Notes
- This PR does not modify the beacon protocol, scheduling, or consensus. It only improves the observability surface of the `/info` endpoint.

AnomalRoil commented:
Thanks, this is a nice addition, but I think it'd be more useful in our https://github.com/drand/http-relay/ repository, since the built-in HTTP server is being deprecated in favor of having less dependencies and a more constrained attack surface in the daemon itself.

alienx5499 commented:
Thanks for the redirect! I've implemented this in the http-relay repository as suggested.

**PR:** https://github.com/drand/http-relay/pull/16

Added `catchup_period_seconds` to the `/info` JSON endpoint. Uses protoreflect for backward compatibility, works with current protobuf versions and will automatically populate once drand protobuf includes the field. Existing clients are unaffected.
